### ISSUE TITLE
Crash with exception Invalid_argument("Incomplete enumeration for the type")

### ISSUE THREAD

**Issue by anne-pacalet** [dbuenzli/cmdliner#114]
I have made some modifications to my options, and now, I get:
```
Fatal error: exception Invalid_argument("Incomplete enumeration for the type")
Raised at file "stdlib.ml", line 30, characters 20-45
Called from file "cmdliner_arg.ml", line 21, characters 21-46
Called from file "camlinternalLazy.ml", line 29, characters 17-27
Re-raised at file "camlinternalLazy.ml", line 36, characters 4-11
Called from file "list.ml", line 103, characters 22-25
Called from file "cmdliner_docgen.ml", line 193, characters 13-66
Called from file "cmdliner_docgen.ml", line 297, characters 33-64
Called from file "cmdliner_docgen.ml", line 308, characters 11-43
Called from file "cmdliner_docgen.ml", line 324, characters 29-42
Called from file "cmdliner_docgen.ml", line 328, characters 46-60
Called from file "cmdliner.ml", line 195, characters 25-71
```
when running `my_command --help`.
It would be better to have some kind of error message...
How can I find out where it comes from?

**Comment by anne-pacalet** [comment]
So, I found the reason, and maybe it could help you to fix the problem.
I had something like:
```
let format = Arg.enum ["csv", Ocsv; "html", Ohtml; ] in
Arg.(value & opt format Opretty & info ["f"; "format"] ~docs ~docv:"FORMAT" ~doc)
```
And I think that the crash was due to not having a string for the default `Opretty` value in `format`.

**Comment by dbuenzli** [comment]
> It would be better to have some kind of error message...

No that's a programming error, it's not error for the end-user. The message could be enhanced though. 

Basically check your usages of [`Arg.enum`](https://erratique.ch/software/cmdliner/doc/Cmdliner.Arg.html#VALenum).

**Comment by anne-pacalet** [comment]
Yes, of course, it was a programming error, but even the programmer needs some messages to understand what is going on ;-)

**Comment by dbuenzli** [comment]
Well you have an error message don't you ?

**Comment by anne-pacalet** [comment]
Yes. It could be clearer though ;-) But you can close the ticket if you wish. Error solved.

**Comment by dbuenzli** [comment]
> Yes. It could be clearer though ;-)

Given the context in which it appears it's not really possible to give a better error message. One can only hint at what to look for. Another option would be not to raise, but that would result in incomplete documentation.

**Comment by anne-pacalet** [comment]
Mmm, I see. Not easy... Thanks for your answers anyway.

**Comment by dbuenzli** [comment]
"Check your `Arg.enum` usages" is already an enhancement over the current message but it's not totally true either. I'll still try to something about it. Thanks for the report.
